client/src/pages/Contact_form.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState, FormEvent } from 'react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import useRecaptcha from '../../useRecaptcha';
+import styles from './Contact_form.module.css'
+
+export const ContactForm = () => {
+  const [name, setName] = useState('');
+  const [surname, setSurName] = useState('');
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState('');
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
+  const [successMessage, setSuccessMessage] = useState('');
+  const { capchaToken, recaptchaRef, handleRecaptcha } = useRecaptcha();
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    setErrors({});
+    setSuccessMessage('');
+
+    const validationErrors: { [key: string]: string } = {};
+
+    if (!name.trim()) validationErrors.name = "Name is required";
+    if (!surname.trim()) validationErrors.surname = "Surname is required";
+    if (!email.includes('@')) validationErrors.email = "Invalid email";
+    if (!message.trim()) validationErrors.message = "The message is mandatory";
+    if (!capchaToken) validationErrors.captcha = "Please validate the reCAPTCHA";
+
+    if (Object.keys(validationErrors).length > 0) {
+      setErrors(validationErrors);
+      return;
+    }
+
+    try {
+      const res = await fetch('/api/contact', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ name, surname, email, message, capchaToken })
+      });
+
+      if (!res.ok) {
+        const { error: msg } = await res.json();
+        setErrors({ global: msg || 'An error has occurred' });
+      } else {
+        setName('');
+        setSurName('');
+        setEmail('');
+        setMessage('');
+        recaptchaRef.current?.reset();
+        setSuccessMessage('Your message has been sent successfully!');
+      }
+    } catch (err) {
+      console.error(err);
+      setErrors({ global: 'A network error has occurred.' });
+    }
+  };
+
+  return (    
+    <form onSubmit={handleSubmit} className= {styles.contactContainer}>
```

Comment:
you can use action instead of onsubmit, so you won't have to implement event on argument and you'll remove the prevent default